src/engine/GroupBy.cpp

```diff
@@ -708,7 +711,7 @@ bool GroupBy::computeOptimizedGroupByIfPossible(IdTable* result) {
 // _____________________________________________________________________________
 std::optional<GroupBy::HashMapOptimizationData>
 GroupBy::checkIfHashMapOptimizationPossible(
-    const std::vector<Aggregate>& aggregates) {
+    std::vector<Aggregate>& aggregates) {
```

Comment:
If it is, then the `SparqlExpresssion` pointers contained in each `aggregate._expression` are `const` as well, and since we pass these into `findAggregateMultiple` and they may be a parent of an aggregate, they cannot be `const`, otherwise we cannot replace one of their children later on.